### PR TITLE
add sand, glass, and calculator

### PR DESCRIPTION
**Pull request by byorgey**
- Sand is now found near water.  Sand regenerates immediately (like water).
- Sand can be used in a furnace to make glass.
- Glass + copper wires + 3D printer can be used to make solar panels.
    - Note that with #361 I intend to start the base with a small number of 3D printers, so other robots could carry this out.  But also, the base will start with a bunch of solar panels too.
- Solar panel + counter can be used to make a calculator.

### CODE REVIEW

**Comment by byorgey** [comment]
Once people are happy with we should also think about whether the tutorial needs to be updated.

**Comment by byorgey** [comment]
At #336 we also discussed the possibility of being able to create sand by drilling rocks; we could easily add that later.

File: data/entities.yaml [hunk 1]
```diff
@@ -378,7 +396,7 @@
 - name: plasma cutter
   display:
     attr: device
-    char: '+'
+    char: '*'
```
Comment (xsebek) [inline data/entities.yaml:399]: This is also a flower, detonator and gear? `P` would make sense as a downward pointing plasma gun :thinking:

**Review by xsebek — APPROVED** [review]
> Once people are happy with we should also think about whether the tutorial needs to be updated.

With what? Otherwise, this looks good to me. :+1:

File: data/entities.yaml [hunk 2]
```diff
@@ -216,6 +216,24 @@
   properties: [portable]
   capabilities: [float]
 
+- name: sand
+  display:
+    attr: copper
+    char: '░'
```
Comment (xsebek) [inline data/entities.yaml:222]: Actually, I can not see the `░` character very well. Maybe its the dirt background and copper attribute?

Comment (xsebek) [reply data/entities.yaml:222]: Oh, the name of the character is "light shade" and my font takes that literally:

image

File: data/entities.yaml (hunk 1 above)
Comment (byorgey) [reply data/entities.yaml:399]: They are all distinguished by color.  But P for plasma cutter sounds good.

File: data/entities.yaml (hunk 2 above)
Comment (byorgey) [reply data/entities.yaml:222]: It's the same character used for dirt, grass, and stone, but should be the same color as copper.  I do want to make sure it's distinguishable in different terminals which don't all handle color the same...

Comment (byorgey) [reply data/entities.yaml:222]: Hmm, in my terminal the copper and dirt colors are easily distinguishable.  But I tried it in gnome-terminal and the two colors are very very hard to tell apart.  Any suggestions for a different color and/or character to use?

Comment (xsebek) [reply data/entities.yaml:222]: I would suggest `IceT` background and the `sand` attribute which I completely forgot about. :smile:

Comment (byorgey) [reply data/entities.yaml]: Hah, didn't know we had a `sand` attribute!  I updated it, how does it look now?

Comment (xsebek) [reply data/entities.yaml]: Looks good enough, in light colour schemes it very much looks like sand for me, in dark ones it's a bit muddy but still distinguishable. :+1:

**Pull request merged**